src/python/WMCore/Services/RucioConMon/RucioConMon.py

```diff
@@ -68,31 +67,26 @@ def _getResult(self, uri, callname="", clearCache=False, args=None, binary=False
         if clearCache:
             self.clearCache(cachedApi, args)
         results = '{}' # explicitly define results which will be loaded by json.loads below
-        if binary:
-            with self.refreshCache(cachedApi, apiUrl, decoder=False, binary=True) as istream:
-                results = gzip.decompress(istream.read())
-            return results
-        else:
-            with self.refreshCache(cachedApi, apiUrl) as istream:
-                results = istream.read()
-
-        results = json.loads(results)
-        return results
+        with self.refreshCache(cachedApi, apiUrl, decoder=True, binary=False) as istream:
+            results = istream.read()
```

Comment:
NOTE: I don't know how much data istream reads, but here you have two potential memory spikes, one is reading data from istream and another to parse the json. Instead, I suggest to use
```
return json.load(istream)
```
which performs both steps as one, i.e. JSON can read from file object directly.